drop-transfer/src/file/mod.rs

```diff
@@ -48,65 +71,65 @@ pub enum FileSource {
     },
 }
 
-#[derive(Clone, Debug)]
-pub struct File {
-    pub(crate) file_id: FileId,
-    pub(crate) subpath: FileSubPath,
-    pub(crate) kind: FileKind,
-}
-
-impl File {
-    fn walk(path: &Path, config: &DropConfig) -> Result<Vec<Self>, Error> {
-        let parent = path
-            .parent()
-            .ok_or_else(|| crate::Error::BadPath("Missing parent directory".into()))?;
-
-        let mut files = Vec::new();
-        let mut breadth = 0;
+impl File for FileToSend {
+    fn id(&self) -> &FileId {
+        &self.file_id
+    }
 
-        for entry in WalkDir::new(path).min_depth(1).into_iter() {
-            let entry = entry?;
-            let meta = entry.metadata()?;
+    fn subpath(&self) -> &FileSubPath {
+        &self.subpath
+    }
 
-            if !meta.is_file() {
-                continue;
-            }
+    fn size(&self) -> u64 {
+        self.meta.len()
+    }
 
-            if entry.depth() > config.dir_depth_limit {
-                return Err(Error::TransferLimitsExceeded);
-            }
+    fn mime_type(&self) -> &str {
+        self.mime_type
+            .as_ref()
+            .map(|x| x.as_str())
+            .unwrap_or(UNKNOWN_STR)
+    }
+}
 
-            breadth += 1;
+impl File for FileToRecv {
+    fn id(&self) -> &FileId {
+        &self.file_id
+    }
 
-            if breadth > config.transfer_file_limit {
-                return Err(Error::TransferLimitsExceeded);
-            }
+    fn subpath(&self) -> &FileSubPath {
+        &self.subpath
+    }
 
-            let subpath = entry
-                .path()
-                .strip_prefix(parent)
-                .map_err(|err| crate::Error::BadPath(err.to_string()))?;
+    fn size(&self) -> u64 {
+        self.size
+    }
 
-            let subpath = FileSubPath::from_path(subpath)?;
+    fn mime_type(&self) -> &str {
+        ""
```

Comment:
we should also return unknown here

Reply:
TBH it is the same as before the refactor. A file to be received was reporting an empty string and a file to be sent reported mime or "unknown".
Here it's the same. Should I change this one?

Reply:
since we discussed constructing the file info for received files, I think it makes sense to use the same logic for both sides

Reply:
In general yes, but with the mime itself, it's a bit different I guess. The thing is you always report "unknown" from receiving side so at least 50% of reported mimes would be "unknown" and the downside is that you can't really tell then in which % of cases the mime is truly unknown (on the sender side).

Reply:
It's a fair point, let's then leave it as it is right and decide if we want to change it when the tracker gets updated